### PR TITLE
Techwebs backend refactor

### PR DESCRIPTION
**Pull request by silicons**
Yeah yeah I know no opening too many WIP PRs but this one's [ready] and I want to make sure any furthur techweb development doesn't keep using the old reference system which will make refactoring it harder, and this makes it easier for coders to do stuff in theory (at least for me), standardizes getting a node/design, and makes stuff actually GC properly.

### CODE REVIEW

**Comment by SpaceManiac** [comment]
is this tested

**Comment by kevinz000** [comment]
@SpaceManiac in like 30 seconds yes

File: code/controllers/subsystem/research.dm [hunk 1]
```diff
@@ -4,18 +4,26 @@ SUBSYSTEM_DEF(research)
 	priority = FIRE_PRIORITY_RESEARCH
 	wait = 10
 	init_order = INIT_ORDER_RESEARCH
+	//TECHWEB STATIC
+	var/list/techweb_nodes = list()				//associative id = node datum
+	var/list/techweb_designs = list()			//associative id = node datum
+	var/list/datum/techweb/techwebs = list()
+	var/datum/techweb/science/science_tech
+	var/datum/techweb/admin/admin_tech
+	var/datum/techweb_node/error_node/error_node//These two are what you get if a node/design is deleted and somehow still stored in a console.
```
Comment (Poojawa) [inline code/controllers/subsystem/research.dm]: need a space fer yer comment

Comment (kevinz000) [reply code/controllers/subsystem/research.dm]: fuck

**Comment by kevinz000** [comment]
alright this is done

**Pull request merged**